### ISSUE TITLE
qwen3-4b-base多机多卡继续预训练时hang住

### ISSUE THREAD

**Issue by yyyhhl** [modelscope/ms-swift#6867]
环境：2机，每台机器2张H20，单机4卡时可以正常训练
命令：
node 0 的命令：
torchrun --nnodes=2 --node_rank=0 --nproc_per_node=2 --master_addr=2605:340:cd51:4b00:9b86:912c:6586:d4d0 --master_port=10217 ./swift_pt.py --model ./workdir/base_model --model_type qwen3 --dataset ./workdir/dataset/shuffled_dataset.jsonl --output_dir ./workdir/swift_output --dataloader_num_workers 10 --dataset_num_proc 10 --attn_impl flash_attn --max_steps 400 --max_length 32768 --per_device_train_batch_size 1 --per_device_eval_batch_size 1 --sequence_parallel_size 1 --learning_rate 1e-5 --warmup_ratio 0.05 --weight_decay 1e-4 --max_grad_norm 1.0 --gradient_accumulation_steps 8 --save_strategy steps --save_steps 400 --save_total_limit 1 --train_type full --deepspeed zero3_offload --eval_strategy steps --eval_steps 10 --split_dataset_ratio 0.05 --lr_scheduler_type warmup_stable_decay --lr_scheduler_kwargs "{ "num_decay_steps": 120, "num_stable_steps": 260}" --packing True --dataset_shuffle True --train_dataloader_shuffle True --truncation_strategy right --dataloader_num_workers 4 --model_author swift --model_name swift-robot --log_level detail --logging_steps 1 --report_to wandb
node 1 的命令：
torchrun --nnodes=2 --node_rank=1 --nproc_per_node=2 --master_addr=2605:340:cd51:4b00:9b86:912c:6586:d4d0 --master_port=10217 ./swift_pt.py --model ./workdir/base_model --model_type qwen3 --dataset ./workdir/dataset/shuffled_dataset.jsonl --output_dir ./workdir/swift_output --dataloader_num_workers 10 --dataset_num_proc 10 --attn_impl flash_attn --max_steps 400 --max_length 32768 --per_device_train_batch_size 1 --per_device_eval_batch_size 1 --sequence_parallel_size 1 --learning_rate 1e-5 --warmup_ratio 0.05 --weight_decay 1e-4 --max_grad_norm 1.0 --gradient_accumulation_steps 8 --save_strategy steps --save_steps 400 --save_total_limit 1 --train_type full --deepspeed zero3_offload --eval_strategy steps --eval_steps 10 --split_dataset_ratio 0.05 --lr_scheduler_type warmup_stable_decay --lr_scheduler_kwargs "{ "num_decay_steps": 120, "num_stable_steps": 260}" --packing True --dataset_shuffle True --train_dataloader_shuffle True --truncation_strategy right --dataloader_num_workers 4 --model_author swift --model_name swift-robot --log_level detail --logging_steps 1 --report_to wandb

在这一步hang住，下面的wandb日志是daemon进程打的，不关键：

<img width="3456" height="1808" alt="Image" src="https://github.com/user-attachments/assets/77d386bd-ec12-41b4-b5f0-cfbcec51f568" />
